### PR TITLE
Stats - Insights - Update Default cards list

### LINKED ISSUE #18042: [iOS] Stats: Insight Defaults View - changes for meaningful cards
As part of Stats revamp:

 * Change the default view in "Insights", which includes:
   * Move "meaningful" cards to the top, so that they are immediately visible to users (TBD exactly which cards, but we seem to have an agreement that views, visitors, likes, comments have priority)

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #18042 

This pull request updates the default Insights screen. This also removes the default "Latest Post Summary" card and removes the "Customize" card from `UserDefaults`

**NOTE: this pull request does not remove the Customize card entirely, just from `UserDefaults`. This is currently being addressed in a separate PR**

To test:
Pre-requisite: have a wp.com account with at least one site (better if more)
Test steps:
- Install a previous version than this branch (e.g. `trunk`)
- Choose a site where you have not customized the Insights screen in Stats
- Go to My Site -> Stats -> Insights
- Verify that the default cards are: "Latest Post Summary", "Today", "All Time" and "Follower Totals", plus any pinned cards (e.g. "Grow Audience or "Customize")
- Build/run this branch
- verify that the default cards now are: "Most Popular Time", "All Time", "Today", "Followers", "Comments"
- Repeat the steps with several sites and observe the behavior: previously customized pages should be left as they were

## Regression Notes
1. Potential unintended areas of impact
not really, this PR only reshuffles default cards in Insights

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test the feature on multiple sites with different scenarios

3. What automated tests I added (or what prevented me from doing so)
None
PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Insights/InsightType.swift
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Insights/InsightType.swift:32]: Shall we also add a comment note here about why this property exists?

**Review by frosty — APPROVED** [review]
Works well for me! Just left one small note.

File: WordPress/Classes/ViewRelated/Stats/Helpers/SiteStatsInformation.swift
Comment (sla8c) [inline WordPress/Classes/ViewRelated/Stats/Helpers/SiteStatsInformation.swift]: First I wanted to say that the desired result works!  I haven't spent as much time in the Insight screens so I had one question about this - can a single  element `InsightType` ever be both of type `.growAudience` AND `.customize`?  From my limited knowledge of Insights, this filter alone would always return an empty result?

**Review by sla8c — APPROVED** [review]
LGTM!  I left one comment as perhaps to help my knowledge of Insights logic.

File: WordPress/Classes/ViewRelated/Stats/Helpers/SiteStatsInformation.swift
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Stats/Helpers/SiteStatsInformation.swift]: uhhhhhh good catch @sla8c !! In fact, this should be an OR. Better, since we are ditching `.customize` based on the discussions in your PR, I'll just update the filter and remove `.customize` altogether!!

File: WordPress/Classes/ViewRelated/Stats/Insights/InsightType.swift
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Stats/Insights/InsightType.swift:32]: for sure!

**Comment by Gio2018** [comment]
@sla8c @frosty I updated according to the comments, and also filtered out `.customize` from `UserDefaults`. This would not yet remove the "Customize" card, but once Steve's [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/18058) gets merged this should match with it. I'll wait a bit in case you want to do a second pass before merging.

**Review by mokagio — DISMISSED** [review]
It looks like the UI tests are not able to deal with the changes in this PR. 

The builds for the latest two commit both failed in stats-related tests.

I hope to be able to get to this later to verify I'm correct. I'm going to "Request changes" to prevent accidental merges in the meantime.

**Comment by mokagio** [comment]
I tried it quickly locally and the failure I got was due to the tests not finding the "35" text in the stats insight screen. 🤔 

https://github.com/wordpress-mobile/WordPress-iOS/blob/3d9700c78fd1dc0ea380212e1bdd26abb73998f4/WordPress/WordPressUITests/Tests/StatsTests.swift#L24

Another thing that might be happening is that the stats insight screen freezes. I experienced this locally but I'm not sure whether it was due to my breakpoints or not and didn't dig further.

Could this be due to the change in default insights to show? Sorry, I haven't had the time to load the app on `trunk` and see what the test does.

**Comment by sla8c** [comment]
Hey @Gio2018 I updated the PR description with the issue I had created for it
https://github.com/wordpress-mobile/WordPress-iOS/issues/18042

Cheers

**Comment by sla8c** [comment]
> I tried it quickly locally and the failure I got was due to the tests not finding the "35" text in the stats insight screen. 🤔
> 
> https://github.com/wordpress-mobile/WordPress-iOS/blob/3d9700c78fd1dc0ea380212e1bdd26abb73998f4/WordPress/WordPressUITests/Tests/StatsTests.swift#L24
> 
> Another thing that might be happening is that the stats insight screen freezes. I experienced this locally but I'm not sure whether it was due to my breakpoints or not and didn't dig further.
> 
> Could this be due to the change in default insights to show? Sorry, I haven't had the time to load the app on `trunk` and see what the test does.

hey @mokagio I'm taking a look now

**Comment by sla8c** [comment]
@mokagio on the latest trunk, the UITest works for me (albeit I didn't have the environment setup to run UITests, so I uninstalled - re-installed WordPress-iOS app trunk build on simulator)

I see the 35 views in simulator + test passes.  s/shots below.  Let me know if I can help anymore! 😄 

image

image

**Comment by Gio2018** [comment]
@mokagio I updated the UI tests reflecting the new default cards. As for the "freeze", if I interpreted correctly, it seemed more like the ghost views would stay on indefinitely (or at least for the test to time out). It happened to me too a couple of times, but could not find any cause yet. Not sure if it was a network lag or something else, and will investigate it, but, most likely, the issue is not directly related to just reshuffling the cards on the screen, so if you'd agree, once the UI tests are resolved, maybe we can merge this one?

**Comment by mokagio** [comment]
@Gio2018 @sla8c thank you so much for taking the time to investigate! 🙇‍♂️ 

> As for the "freeze", if I interpreted correctly, it seemed more like the ghost views would stay on indefinitely (or at least for the test to time out). It happened to me too a couple of times, but could not find any cause yet. Not sure if it was a network lag or something else, and will investigate it, but, most likely, the issue is not directly related to just reshuffling the cards on the screen, so if you'd agree, once the UI tests are resolved, maybe we can merge this one?

I agree the issue is unlikely to be related with the changes in this PR. Networking or mocks sound like the most likely culprit to me, too—or plain old "UI tests are weird".

I'll dismiss my review and keep and eye on this. Once CI is green, I'll merge and ship a new build.

Thanks again! 🎉

**Comment by mokagio** [comment]
@Gio2018 the UI tests on iPad were still red, with stats-related issue and now the build fails on `release/19.4`, too.

image

I'm looking at [in a branch off this one](https://buildkite.com/automattic/wordpress-ios/builds/5773#da27d801-82d6-41f5-9511-5a01db45b3e6). I'm confident it'll be something minor that needs fixing, it's just time consuming work because of switching the pods etc. 😩

**Comment by Gio2018** [comment]
> @Gio2018 the UI tests on iPad were still red, with stats-related issue and now the build fails on `release/19.4`, too.
> 
> image
> 
> I'm looking at [in a branch off this one](https://buildkite.com/automattic/wordpress-ios/builds/5773#da27d801-82d6-41f5-9511-5a01db45b3e6). I'm confident it'll be something minor that needs fixing, it's just time consuming work because of switching the pods etc. 😩

@mokagio I can't wrap my head around this. Looking at the logs, it says that it can't find "Views", but it's obviously there and it succeeded several times locally, not to mention that on iPhone succeeds. Would it be an option to temporarily disable this particular test until we find what's wrong?

**Comment by mokagio** [comment]
Yeah, me neither. On my machine, it passes on the first run on the same Simulator used in CI. But something else weird happens... The test then run again, and there it fails. Twice.

I'm trying to get CI to take a screenshot of the failure (which it should already do but somehow doesn't) to see if we can get more info.

> Would it be an option to temporarily disable this particular test until we find what's wrong?

Yes. I'm gonna keep at it in my today then maybe ask @tiagomar if he has time to look into this too. I'll have a PR ready to make this an expected failure, so if he doesn't have time or cannot find an answer in reasonable time, we can move forward.

**Comment by Gio2018** [comment]
> Yeah, me neither. On my machine, it passes on the first run on the same Simulator used in CI. But something else weird happens... The test then run again, and there it fails. Twice.
> 
> I'm trying to get CI to take a screenshot of the failure (which it should already do but somehow doesn't) to see if we can get more info.
> 
> > Would it be an option to temporarily disable this particular test until we find what's wrong?
> 
> Yes. I'm gonna keep at it in my today then maybe ask @tiagomar if he has time to look into this too. I'll have a PR ready to make this an expected failure, so if he doesn't have time or cannot find an answer in reasonable time, we can move forward.

Sounds good. I'll also look better tomorrow; I've been looking into this a little more and, can't still picture exactly what, but something smells here.. @sla8c can you pick this up since you're starting before me? I'd check if there's anything strange with the state changes due to loading from the network or the cache, in `StatsInsightStore`. Thanks so much @mokagio !

**Comment by mokagio** [comment]
> I'm trying to get CI to take a screenshot of the failure (which it should already do but somehow doesn't) to see if we can get more info.

Update: It was actually me not looking at the `.xcresult` right 🤦‍♂️ 

CI repeats the test two times after the first failure. This is the screen during the first failure:

image

The following retries fail with the screen in this state:

image

**Comment by tiagomar** [comment]
The test is looking for the `ALL-TIME` which was actually loaded right after the login, according to `.xcresult`.

image

[I have worked recently on a similar issue](https://github.com/wordpress-mobile/WordPress-iOS/pull/18008), the difference is just that in that case there was an error message. The issue happens when `Stats` button is tapped, what we can also see from the images in @mokagio's [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/18072#issuecomment-1061369423). 

image

The workaround was to add a `pull to refresh` when the error message is present `refreshStatsIfNeeded()` and all the data would load properly. Since the error message is no longer present now, the PTR was not triggered. I wasn't aware of changes in caching strategy when this test started failing consistently. The good part is that the test did its job and identified the change. =P 

So maybe it is time to have it matching whatever changes we have made.

**Comment by mokagio** [comment]
> The good part is that the test did its job and identified the change.

🎉 🎉 🎉

**Comment by mokagio** [comment]
@Gio2018  this has been bundled as part of [19.4 beta 1 (19.4.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/18115).

Thanks for your work 🙌

**Pull request merged**
